mango/clients/aemet.py

```diff
@@ -0,0 +1,598 @@
+
+    :param api_key: API key to connect to the AEMET API
+    :param wait_time: Wait time between requests in seconds. Default is 1.25 seconds
+    :doc-author: baobab soluciones
+
+    Usage
+    -----
+
+    >>> import load_dotenv
+    >>> import os
+    >>> from mango.clients.aemet import AemetClient
+    >>> load_dotenv()
+    >>> client = AemetClient(os.environ["AEMET_API_KEY"])
+    """
+
+    def _set_up_default(self) -> None:
+        """
+        Do as function to allow for updates mid execution and set up default values
+        """
+        self._DEFAULT_WAIT_TIME = float(os.environ.get("AEMET_DEFAULT_WAIT_TIME", 1.25))
+        self._FETCH_STATIONS_URL = os.environ.get(
+            "AEMET_FETCH_STATIONS_URL",
+            "https://opendata.aemet.es/opendata/api/valores/climatologicos/inventarioestaciones/todasestaciones",
+        )
+        self._FETCH_MUNICIPIOS_URL = os.environ.get(
+            "AEMET_FETCH_MUNICIPIOS_URL",
+            "https://opendata.aemet.es/opendata/api/maestro/municipios",
+        )
+        self._FETCH_HISTORIC_URL = os.environ.get(
+            "AEMET_FETCH_HISTORIC_URL",
+            "https://opendata.aemet.es/opendata/api/valores/climatologicos/diarios/datos/fechaini"
+            "/{start_date_str}/fechafin/{end_date_str}"
+            "/estacion/{station}",
+        )
+        self._FETCH_DAILY_URL = os.environ.get(
+            "AEMET_FETCH_DAILY_URL",
+            "https://opendata.aemet.es/opendata/api/observacion/convencional/datos/estacion/{station}",
+        )
+        self._FETCH_FORECAST_URL = os.environ.get(
+            "AEMET_FETCH_FORECAST_URL",
+            "https://opendata.aemet.es/opendata/api/prediccion/especifica/municipio/diaria/{postal_code}",
+        )
+
+    def __init__(self, *, api_key: str, wait_time: Union[float, int] = None):
+        super().__init__()
+        self._set_up_default()
+        self._api_key = api_key
+        self._wait_time = wait_time or self._DEFAULT_WAIT_TIME
+        logging.getLogger("root")
+        if self._wait_time < self._DEFAULT_WAIT_TIME:
+            logging.warning(
+                "Wait time is too low. API limits may be exceeded. Minimum wait time is {} seconds.".format(
+                    self._DEFAULT_WAIT_TIME
+                )
+            )
+        logging.info(
+            "Aemet API has a limit of 50 requests per minute. Please be patient. Wait time is set to {} seconds.".format(
+                self._wait_time
+            )
+        )
+
+    def connect(self) -> None:
+        """
+        This method will connect to the AEMET API and cache the stations and municipios.
+        Checks if the API key is valid.
+        :doc-author: baobab soluciones
+        """
+        # Allows to check if the api_key is valid and cache the stations
+        try:
+            self._all_stations = self._get_all_stations()
+        except ApiKeyError as e:
+            raise e
```

Comment:
Do they not have any kind of health endpoint?

Reply:
Not that I am aware of (https://opendata.aemet.es/dist/index.html#/) When I added this seemed like a good trade-off to check for api key validation as the stations info is needed for other queries and doesn´t take too much